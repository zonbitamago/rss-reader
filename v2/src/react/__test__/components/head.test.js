'user strict';
jest.dontMock('../../components/head.jsx');
import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import Head from '../../components/head.jsx';

const onHomeClick = jest.fn();

const actions = {
  onHomeClick: onHomeClick
};

test('snapshot', function() {
  const tree = renderer.create(<Head actions={actions}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('onClick home',() => {
  const head = mount(<Head actions={actions}/>);
  head.find('.home.large.icon').simulate('click');
  expect(onHomeClick).toHaveBeenCalled();
})
