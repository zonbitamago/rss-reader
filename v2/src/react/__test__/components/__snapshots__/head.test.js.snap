// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="ui blue inverted pointing secondary head menu"
>
  <a
    className="link item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="home large icon"
    />
  </a>
  <div
    className="item"
    onClick={[Function]}
  >
    <h3
      className="ui inverted header"
    >
      Home
    </h3>
  </div>
</div>
`;
